apollo-router/src/query_planner/fetch.rs

```diff
@@ -627,13 +627,18 @@ impl FetchNode {
 
     pub(crate) fn extract_authorization_metadata(
         &mut self,
-        subgraph_schemas: &SubgraphSchemas,
+        schema: &Valid<apollo_compiler::Schema>,
         global_authorisation_cache_key: &CacheKeyMetadata,
     ) {
-        let doc = self.parsed_operation(subgraph_schemas);
-        let schema = &subgraph_schemas[self.service_name.as_str()];
+        let doc = ExecutableDocument::parse(
```

Comment:
Re parsing here is unfortunate. Would it make sense to call `self.parsed_operation()` here with the subgraph schema, but then pass the supergraph schema to `AuthorizationPlugin::generate_cache_metadata` for finding directives?

Reply:
This was resolved by off-GH discussion. I'm going to merge.